### PR TITLE
Many code optimizations (performance, style, comments, etc.)

### PR DESCRIPTION
**Pull request by aarongerig**

### CODE REVIEW

File: src/ImportDefinitionsBundle/Cleaner/ReferenceCleaner.php [hunk 1]
```diff
@@ -30,7 +31,7 @@ public function cleanup(DefinitionInterface $definition, $objects)
             $dependency = $obj->getDependencies();
 
             if ($dependency instanceof Dependency) {
-                if (count($dependency->getRequiredBy()) === 0) {
+                if (\count($dependency->getRequiredBy()) === 0) {
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Cleaner/ReferenceCleaner.php]: why?

File: src/ImportDefinitionsBundle/Form/Type/Interpreter/CoreShop/StoresInterpreterType.php [hunk 2]
```diff
@@ -23,25 +23,25 @@
...
                 },
                 function ($value) {
                     $resolvedValues = [];
-                    if (is_array($value) && array_key_exists('stores', $value)) {
-                        if ($value['stores'] instanceof ArrayCollection) {
-                            foreach ($value['stores'] as $val) {
-                                if ($val instanceof StoreInterface) {
-                                    $resolvedValues[] = $val->getId();
-                                }
+                    if (\is_array($value) && array_key_exists('stores', $value) && $value['stores'] instanceof ArrayCollection) {
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Form/Type/Interpreter/CoreShop/StoresInterpreterType.php]: not quite pretty though. maybe something like

```
if (!is_array($value)  {
return;
}
if (!array_key_exists('stores', $value)) {
return;
}
....
```

File: src/ImportDefinitionsBundle/Form/Type/Setter/CoreShop/StorePriceSetterType.php [hunk 3]
```diff
@@ -36,12 +35,10 @@ function ($value) {
                 },
                 function ($value) {
                     $resolvedValues = [];
-                    if (is_array($value) && array_key_exists('stores', $value)) {
-                        if ($value['stores'] instanceof ArrayCollection) {
-                            foreach ($value['stores'] as $val) {
-                                if ($val instanceof StoreInterface) {
-                                    $resolvedValues[] = $val->getId();
-                                }
+                    if (\is_array($value) && array_key_exists('stores', $value) && $value['stores'] instanceof ArrayCollection) {
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Form/Type/Setter/CoreShop/StorePriceSetterType.php]: same as above

File: src/ImportDefinitionsBundle/Importer/Importer.php [hunk 4]
```diff
@@ -131,31 +132,33 @@ public function doImport(DefinitionInterface $definition, $params)
...
+        if ($filterType) {
+            $filter = $this->filterRegistry->get($filterType);
         }
 
         $data = $this->getData($definition, $params);
 
-        if (count($data) > 0) {
-            $this->eventDispatcher->dispatch('import_definition.total', new ImportDefinitionEvent($definition, count($data)));
+        if (\count($data) > 0) {
+            $this->eventDispatcher->dispatch('import_definition.total', new ImportDefinitionEvent($definition, \count($data)));
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Importer/Importer.php]: why `\count` ?

File: src/ImportDefinitionsBundle/Importer/Importer.php [hunk 5]
```diff
@@ -210,12 +217,12 @@ protected function getData(DefinitionInterface $definition, $params)
...
      * @throws \Exception
      */
-    protected function runImport(DefinitionInterface $definition, $params, $filter = null, $data = array())
+    private function runImport(DefinitionInterface $definition, $params, $filter = null, array $data = [])
     {
         $count = 0;
         $countToClean = 1000;
 
-        if (is_array($data)) {
+        if (\is_array($data)) {
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Importer/Importer.php]: why `\is_array`?

File: src/ImportDefinitionsBundle/Importer/Importer.php [hunk 6]
```diff
@@ -379,21 +382,21 @@ protected function getObjectForPrimaryKey(DefinitionInterface $definition, $data
...
             $list->load();
             $objectData = $list->getObjects();
 
-            if (count($objectData) === 1) {
+            if (\count($objectData) === 1) {
                 $obj = $objectData[0];
             }
 
-            if (!isset($obj)) {
+            if ($obj === null) {
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Importer/Importer.php]: according to symfony docs it should be a yoda expression `null === $obj`

File: src/ImportDefinitionsBundle/Interpreter/CheckboxInterpreter.php [hunk 7]
```diff
@@ -25,7 +25,8 @@ class CheckboxInterpreter implements InterpreterInterface
      */
     public function interpret(Concrete $object, $value, Mapping $map, $data, DefinitionInterface $definition, $params, $configuration)
     {
-        $boolVal = is_string($value) ? filter_var($value, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE) : (bool)$value;
-        return ($boolVal === null) ? false : $boolVal;
+        $boolVal = \is_string($value) ? filter_var($value, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE) : (bool) $value;
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Interpreter/CheckboxInterpreter.php]: why `\is_string`?

File: src/ImportDefinitionsBundle/Model/Definition/Dao.php [hunk 8]
```diff
@@ -129,7 +124,7 @@ public function save()
                 'failureNotificationDocument', 'successNotificationDocument', 'skipExistingObjects', 'skipNewObjects'];
 
             foreach ($dataRaw as $key => $value) {
-                if (in_array($key, $allowedProperties)) {
+                if (\in_array($key, $allowedProperties, true)) {
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Model/Definition/Dao.php]: why `\in_array`?

File: src/ImportDefinitionsBundle/Model/Definition/Dao.php [hunk 9]
```diff
@@ -138,7 +133,7 @@ public function save()
                         if ($value) {
                             $data[$key] = array();
 
-                            if (is_array($value)) {
+                            if (\is_array($value)) {
                                 foreach ($value as $index => $map) {
                                     $data[$key][$index] = get_object_vars($map);
                                 }
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Model/Definition/Dao.php]: why `\in_array`?

File: src/ImportDefinitionsBundle/Model/Definition/Listing/Dao.php [hunk 10]
```diff
@@ -48,15 +49,15 @@ public function load()
...
+     * @throws \Exception
      */
-    public function getTotalCount()
+    public function getTotalCount(): int
     {
         $data = $this->db->fetchAll($this->model->getFilter(), $this->model->getOrder());
-        $amount = count($data);
 
-        return $amount;
+        return \count($data);
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Model/Definition/Listing/Dao.php]: why `\count`?

File: src/ImportDefinitionsBundle/Model/Log/Dao.php [hunk 11]
```diff
@@ -54,30 +54,30 @@ public function save()
...
-                if (!in_array($k, $validColumns)) {
+                if (!\in_array($k, $validColumns, true)) {
                     continue;
                 }
 
-                $getter = "get" . ucfirst($k);
+                $getter = sprintf('get%s', ucfirst($k));
 
-                if (!is_callable([$this->model, $getter])) {
+                if (!\is_callable([$this->model, $getter])) {
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Model/Log/Dao.php]: why `\is_callable`?

File: src/ImportDefinitionsBundle/Model/Log/Dao.php [hunk 12]
```diff
@@ -54,30 +54,30 @@ public function save()
...
                     continue;
                 }
 
                 $value = $this->model->$getter();
 
-                if (is_bool($value)) {
-                    $value = (int)$value;
+                if (\is_bool($value)) {
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Model/Log/Dao.php]: why `\is_bool`?

File: src/ImportDefinitionsBundle/Provider/CsvProvider.php [hunk 13]
```diff
@@ -84,16 +83,17 @@ public function getData($configuration, $definition, $params, $filter = null)
...
         }
 
         $data = [];
 
         $row = 0;
-        if (($handle = fopen($file, "r")) !== false) {
-            while (($csvData = fgetcsv($handle, 1000, $delimiter, $enclosure ? $enclosure : chr(8))) !== false) {
-                $num = count($csvData);
+        if (($handle = fopen($file, 'rb')) !== false) {
+            while (($csvData = fgetcsv($handle, 1000, $delimiter, $enclosure ?: \chr(8))) !== false) {
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Provider/CsvProvider.php]: why `\chr`?

File: src/ImportDefinitionsBundle/Provider/CsvProvider.php [hunk 14]
```diff
@@ -84,16 +83,17 @@ public function getData($configuration, $definition, $params, $filter = null)
...
+                $num = \count($csvData);
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Provider/CsvProvider.php]: why `\count`?

File: src/ImportDefinitionsBundle/Form/Type/Interpreter/CoreShop/StoresInterpreterType.php (hunk 2 above)
Comment (aarongerig) [reply src/ImportDefinitionsBundle/Form/Type/Interpreter/CoreShop/StoresInterpreterType.php]: What about...

```php
if (!\is_array($value) ||
    !array_key_exists('stores', $value) ||
    !$value['stores'] instanceof ArrayCollection) {
    return [];
}
    
foreach ($value['stores'] as $val) {
    if ($val instanceof StoreInterface) {
        $resolvedValues[] = $val->getId();
    }
}
```

?

Comment (dpfaffenbauer) [reply src/ImportDefinitionsBundle/Form/Type/Interpreter/CoreShop/StoresInterpreterType.php]: 👍

File: src/ImportDefinitionsBundle/Importer/Importer.php [hunk 15]
```diff
@@ -396,7 +396,7 @@ private function getObjectForPrimaryKey(DefinitionInterface $definition, $data)
                 $obj = $objectData[0];
             }
 
-            if ($obj === null) {
+            if (null === $obj) {
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Importer/Importer.php]: may the force be with you my young Padawan

Comment (aarongerig) [reply src/ImportDefinitionsBundle/Importer/Importer.php]: Always master!

**Comment by dpfaffenbauer** [comment]
cheers :)

**Pull request merged**